### PR TITLE
Move to react version 6

### PR DESCRIPTION
**Pull request by karelhala**
There's fairly big braking change in react-router-dom v6. We have to start using it soon, otherwise we'd end with a lot of bugs and harder upgrade path.

### CODE REVIEW

File: frontend/src/Routes/DynamicRoute/DynamicRoute.tsx [hunk 1]
```diff
@@ -21,6 +21,7 @@ const DynamicRoute: React.FC<DynamicRouteProps> = () => {
   const routes = React.useMemo(
     () =>
       dynamicRoutePages.map(({ properties: { component, ...currRoute }, pluginName, uid }) => {
+        delete currRoute.exact;
```
[the remark this answers is not in the document]
Comment (karelhala) [reply frontend/src/Routes/DynamicRoute/DynamicRoute.tsx:24]: Yeah, good point. Just a question, can plugin have its own router?

[the remark this answers is not in the document]
Comment (karelhala) [reply frontend/src/Routes/DynamicRoute/DynamicRoute.tsx]: While doing this I've also added backwards compatibility for array routes.

[the remark this answers is not in the document]
Comment (christianvogt) [reply frontend/src/Routes/DynamicRoute/DynamicRoute.tsx]: A plugin can certainly do its own thing w.r.t routing in its nested components.

**Comment by christianvogt** [comment]
lgtm, but i haven't run the change

**Comment by florkbr** [comment]
Tested the change locally and things seem to function as intended WRT Core and Dev :shipit:

File: frontend/src/Routes.tsx [hunk 2]
```diff
@@ -1,10 +1,9 @@
 import * as React from 'react';
-import { Redirect, Route, Switch } from 'react-router-dom';
+import { Route, Routes as DomRoutes } from 'react-router-dom';
 
 import { Bullseye, Spinner } from '@patternfly/react-core';
 
 const DynamicRoute = React.lazy(() => import(/* webpackChunkName: "DynamicRoute" */ './Routes/DynamicRoute/DynamicRoute'));
-const EmptyRoute = React.lazy(() => import(/* webpackChunkName: "EmptyRoute" */ './Routes/EmptyRoute'));
```
[the remark this answers is not in the document]
Comment (karelhala) [reply frontend/src/Routes.tsx:7]: Yeah, let's remove the empty page in a followup PR.

**Pull request merged**